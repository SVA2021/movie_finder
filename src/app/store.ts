import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import authReducer from '../features/auth/authSlice';
import movieFinderReducer from '../features/movieFinder/movieFinderSlice';

export const store = configureStore({
  reducer: {
    auth: authReducer,
		movieFinder: movieFinderReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
