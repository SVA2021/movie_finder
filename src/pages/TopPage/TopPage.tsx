import {Box} from "@mui/material";
import {FC} from 'react';
import {useParams} from "react-router-dom";
import {useAppDispatch, useAppSelector} from "../../app/hooks";
import {ExpandableList} from "../../components";
import {selectHomePage} from "../../features/movieFinder/movieFinderSlice";
import {getTopListAsync} from "../../features/movieFinder/movieFinderThunks";
import {getTopListTitle, getTopListType} from "../../utils";

export const TopPage: FC = () => {

  const params = useParams();
  const type = getTopListType(params.type);
  const dispatch = useAppDispatch();
  const HomePage = useAppSelector(selectHomePage);
  const totalPages = HomePage[type]?.pagesCount ?? 0;
  const currentPage = HomePage[type]?.page ?? 1;

  function addNewPage() {
    dispatch(getTopListAsync({type, page: currentPage + 1}))
  }

  return (
    <Box pt={1} data-testid={'top'}>
      <ExpandableList
        title={getTopListTitle(type)}
        data={HomePage[type]?.data ?? []}
        page={currentPage}
        totalPages={totalPages}
        addPage={() => addNewPage()}
      />
    </Box>
  )
};