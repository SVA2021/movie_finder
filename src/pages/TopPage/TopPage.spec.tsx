import {render, screen} from "@testing-library/react";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import {store} from "../../app/store";
import {TopPage} from "./TopPage";

describe('render TopPage', () => {

  const setUp = () => render(
    <Provider store={store}>
      <BrowserRouter>
        <TopPage />
      </BrowserRouter>
    </Provider>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('top')).toMatchSnapshot();
  });

})