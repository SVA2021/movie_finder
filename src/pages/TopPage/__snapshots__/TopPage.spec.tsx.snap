// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render TopPage snapshot test 1`] = `
<div
  class="MuiBox-root css-zoser8"
  data-testid="top"
>
  <div
    class="MuiBox-root css-zoser8"
    data-testid="expandable-list"
  >
    <h2
      class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-9q8iy8-MuiTypography-root"
    >
      100 самых популярных фильмов
    </h2>
    <p
      class="MuiTypography-root MuiTypography-h3 css-13mkr37-MuiTypography-root"
      data-testid="empty-data-box"
    >
      Нет данных
    </p>
  </div>
</div>
`;
